applications/workspaces/server/workspaces/service/model_service.py

```diff
@@ -0,0 +1,144 @@
+import json
+from cloudharness import log as logger
+
+from cloudharness.events.decorators import send_event
+
+import cloudharness.workflows.argo as argo
+
+from workspaces.models import OSBRepository, OSBRepositoryEntity
+from workspaces.repository import (
+    OSBRepositoryRepository,
+    VolumeStorageRepository,
+    WorkspaceRepository,
+    WorkspaceResourceRepository,
+)
+
+
+
+def rm_null_values(dikt):
+    tmp = {}
+    for k, v in dikt.items():  # remove null fields from dict
+        if v:
+            tmp.update({k: v})
+    return tmp
+
+
+class BaseModelService():
+    """
+    Generic base model service class
+    """
+
+    repository = None
+
+    def search(self, page=1, per_page=20, *args, **kwargs):
+        """
+        Query the model and return all records
+
+        Args:
+            page: The page number
+            per_page: The number of records per page to limit
+
+        Returns:
+            dict with all records from the model
+            current page
+            number of pages
+        """
+        logger.debug("Search args %s", args)
+        logger.debug("Search kwargs %s", kwargs)
+        return self.repository.search(page=page, per_page=per_page, *args, **kwargs)
+
+    def post(self, body):
+        """Save an object to the repository."""
+        body = rm_null_values(body)
+        return self.repository.post(body)
+
+    def get(self, id_):
+        """Get an object from the repository."""
+        obj = self.repository.get(id=id_)
+        if obj is None:
+            return f"{self.repository.model.__name__} with id {id_} not found.", 404
```

Comment:
the service layer shouldn't be aware of the view layer

Reply:
agree, this is obsolete code, I removed it in commit https://github.com/OpenSourceBrain/OSBv2/pull/231/commits/5baa93dc025b403b2753711daac42ed967475949